src/fenicsxconcrete/finite_element_problem/concrete_am.py

```diff
@@ -0,0 +1,506 @@
+        Returns:
+            array of pseudo density
+        """
+
+        l_active = np.zeros_like(path_time)  # 0: non-active
+
+        activ_idx = np.where(path_time >= 0 - 1e-5)[0]
+        l_active[activ_idx] = 1.0  # active
+
+        return l_active
+
+    @staticmethod
+    def E_fkt(pd: float, path_time: float, parameters: dict) -> float:
+        """computes the Young's modulus at current quadrature point according to bilinear Kruger model
+
+        Args:
+            pd: value of pseudo density [0 - non active or 1 - active]
+            path_time: process time value
+            parameters: parameter dict for bilinear model described by (P0,R_P,A_P,tf_P,age_0)
+
+        Returns:
+            value of current Young's modulus
+        """
+        # print(parameters["age_0"] + path_time)
+        if pd > 0:  # element active, compute current Young's modulus
+            age = parameters["age_0"] + path_time  # age concrete
+            if age < parameters["tf_P"]:
+                E = parameters["P0"] + parameters["R_P"] * age
+            elif age >= parameters["tf_P"]:
+                E = (
+                    parameters["P0"]
+                    + parameters["R_P"] * parameters["tf_P"]
+                    + parameters["A_P"] * (age - parameters["tf_P"])
+                )
+        else:
+            E = 1e-4  # non-active
+
+        return E
+
+
+class ConcreteThixElasticModel(df.fem.petsc.NonlinearProblem):
+    """linear elastic thixotropy concrete model
+
+        linear elasticity law with age dependent Youngs modulus modelling the thixotropy
+        tensor format!!
+
+    Args:
+        mesh : The mesh.
+        parameters : Dictionary of material parameters.
+        rule: The quadrature rule.
+        u: displacement fct
+        bc: array of Dirichlet boundaries
+        body_force: function of creating body force
+
+    """
+
+    def __init__(
+        self,
+        mesh: df.mesh.Mesh,
+        parameters: dict[str, int | float | str | bool],
+        rule: QuadratureRule,
+        u: df.fem.Function,
+        bc: list[df.fem.DirichletBCMetaClass],
+        body_force_fct: Callable,
+    ):
+
+        self.p = parameters
+        self.rule = rule
+        self.mesh = mesh
+        dim_to_stress_dim = {1: 1, 2: 4, 3: 9}  # Tensor formulation!
+        self.stress_strain_dim = dim_to_stress_dim[self.p["dim"]]
+
+        # generic quadrature function space
+        q_V = self.rule.create_quadrature_space(self.mesh)
+        q_VT = self.rule.create_quadrature_tensor_space(self.mesh, (self.p["dim"], self.p["dim"]))
+
+        # quadrature functions (required in pde)
+        self.q_E = df.fem.Function(q_V, name="youngs_modulus")
+        self.q_fd = df.fem.Function(q_V, name="density_increment")
+
+        # path variable from AM Problem
+        self.q_array_path = self.rule.create_quadrature_array(self.mesh, shape=1)
+        self.q_array_path[:] = 0.0  # default all active set by "set_initial_path(q_array_path)"
```

Comment:
`create_quadrature_array` already returns zeros, but for readability we can keep the line if you want to

```suggestion

```